src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/Service.cs

```diff
@@ -0,0 +1,399 @@
+            NNStreamerError ret;
+
+            if (is_input)
+                ret = Interop.Service.GetInputInformation(_handle, name, out inHandle);
+            else
+                ret = Interop.Service.GetOutputInformation(_handle, name, out inHandle);
+
+            NNStreamer.CheckException(ret, "Failed to get TensorsInfo handle");
+
+            TensorsInfo retInfo = TensorsInfo.ConvertTensorsInfoFromHandle(inHandle);
+            return retInfo;
+        }
+
+        public void SetInformation(string name, string value)
+        {
+            if (string.IsNullOrEmpty(name))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property name is invalid");
+
+            if (string.IsNullOrEmpty(value))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property value is invalid");
+
+            NNStreamerError ret = Interop.Service.SetInformation(_handle, name, value);
+            NNStreamer.CheckException(ret, "Failed to set service information");
+        }
+
+        public string GetInformation(string name)
+        {
+            if (string.IsNullOrEmpty(name))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property name is invalid");
+
+            string value;
+            NNStreamerError ret = Interop.Service.GetInformation(_handle, name, out value);
+            NNStreamer.CheckException(ret, "Failed to get service information");
+
+            return value;
+        }
+
+        public void Request(string name, TensorsData data)
+        {
+            if (string.IsNullOrEmpty(name))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property name is invalid");
+
+            if (data == null)
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "Given data is invalid");
+
+            NNStreamerError ret = Interop.Service.Request(_handle, name, data.GetHandle());
+            NNStreamer.CheckException(ret, "Failed to request service");
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+                return;
+
+            if (disposing)
+            {
+                // release managed object
+            }
+
+            // release unmanaged objects
+            if (_handle != IntPtr.Zero)
+            {
+                NNStreamerError ret = Interop.Service.Destroy(_handle);
+                if (ret != NNStreamerError.None)
+                    Log.Error(NNStreamer.TAG, "Failed to destroy the service handle");
+
+                _handle = IntPtr.Zero;
+            }
+
+            _disposed = true;
+        }
+
+        static public void SetPipeline(string name, string desc)
```

Comment:
Suggestion: need discussion for name rule with inner class, then we can call method
Service.Pipeline.Set(name, desc)
Service.Model.Register(name, path, ...)
```
public class Pipeline
{
  internal Pipeline()
  {
  }

  static public void Set(name, desc)
  static public string Get(name)
}
```

Reply:
Used inner class for static functions :)